File: programmers/코딩테스트 입문 with Java/개미군단.java
package 코딩테스트입문;

/**
 * https://school.programmers.co.kr/learn/courses/30/lessons/120837
 */
public class 개미군단 {
    public static void main(String[] args) {
        System.out.println(solution(23));
        System.out.println(solution(24));
        System.out.println(solution(999));
        System.out.println(solution(25));
        System.out.println(solution(9));
        System.out.println(solution(2));
    }

    public static int solution(int hp) {
        int answer = 0;
        if (hp < 3) {
            return hp;
        }
        answer += (hp / 5);

        hp %= 5;
        if (hp > 3) answer += 2;
        else if (hp > 0) answer++;
        return answer;
    }

}

File: programmers/코딩테스트 입문 with Python/개미군단.py
def solution(hp):
    answer = 0
    a, b = divmod(hp, 5)
    answer += a
    if b != 0:
        c, d = divmod(b, 3)
        answer += (c + d)

    return answer


print(solution(23))
print(solution(24))
print(solution(999))
